src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  const objsAfterActions = [];
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    if (type === 'addProperties') {
+      stateCopy = { ...stateCopy };
```

Comment:
You don't need to create a new copy of the state here, as you're already creating a copy before entering the loop on line 5.